frontend/test/metabase/visualizations/lib/fill_data.unit.spec.js

```diff
@@ -0,0 +1,59 @@
+import moment from "moment";
+
+import fillMissingValuesInDatas from "metabase/visualizations/lib/fill_data";
+
+describe("fillMissingValuesInDatas", () => {
+  it("should fill missing days", () => {
+    const time1 = moment("2018-01-01");
+    const time2 = moment("2018-01-31");
+    const rows = [[time1, 1], [time2, 2]];
+    const [filledData] = fillMissingValuesInDatas(
+      {
+        series: [{}],
+        settings: {
+          "graph.x_axis.scale": "timeseries",
+          series: () => ({ "line.missing": "none" }),
+        },
+      },
+      {
+        xValues: [time1, time2],
+        xDomain: [time1, time2],
+        xInterval: { interval: "day", count: 1 },
+      },
+      [rows],
+    );
+
+    expect(filledData.length).toEqual(31);
+    expect(filledData[1][1]).toEqual(null);
+  });
+
+  it("should work with data in a different timezone", () => {
```

Comment:
One issue we've run into is timezones with daylights savings and data that spans DST and non-DST days. We should have a test for that.

We might need to finally finish this PR to get this to work correctly https://github.com/metabase/metabase/pull/7247

Reply:
Good thinking! I just pushed an additional test for this. It worked already because of how the offset gets overwritten.